<p><b>NAME:</b> List.</p><p><b>ALSO KNOWN AS:</b> Linked list.</p><p><b>CLASSIFICATION:</b> Domain Modelling.</p><p><b>MOTIVATION:</b> An ordered group of elements is a very intuitive modelling structure, yet the semantics of such a construct in OWL DL are complex. Biology is full of structures where the order of the elements is vital (e.g. parts of genes). If that order is altered (e.g. a change of the order of introns and exons in a gene) there can be serious damage in biological systems.</p><p><b>AIM:</b> The List is used to model ordered elements, representing the semantics of the order: in this case the ODP will be used to build a gene starting from some elements of the Sequence Ontology: Promoter (SO:0000167), Terminator (SO:0000141), Intron (SO:0000188) and Exon (SO:0000147). For the sake of clarity a minimalist gene is built, with a very simple structure.</p><p><b>STRUCTURE:</b></p><img src="../img/List_abstract.png"><p><b>SAMPLE:</b></p><img src="../img/List_instance.png"><p><b>ELEMENTS:</b> The most important elements are the different classes that can be used to build the List (Promoter, Terminator, Intron and Exon) and the class that it is modelled using the List (in this case Gene). The needed relationships are: Contents (functional), Rest (transitive) and Next (functional and a subproperty of Rest).</p><p><b>IMPLEMENTATION:</b> There is a Protege wizard available for easily creating lists.</p><p><b>RESULT:</b> The result is the class Gene, with the elements in the proper order. Apart of being an efficient way of modelling ordered elements, Lists offer the possibility of creating a powerful classyfing system: Lists of plenty of kinds can be defined (e.g. definitions of the following type: any List containing elements A and B, not followed by C and then followed by two D-s) and they will be put in the correct position of the hierarchy of already defined lists. Using that procedure, for example, different protein fingerprints (lists of regular expressions) or different kinds of genes can be defined. The models can be queried, for example, with a given gene defined with a certain ordered combination of introns, exons, promoter and terminator to see in which position of the hierarchy is classified and to which genes does it relate.</p><p><b>SIDE EFFECTS:</b> (i) If very long and complex lists are used there can be a decrease in reasoning performance. (ii) Maintenance of Lists is a very difficult task.</p><p><b>ADDITIONAL INFORMATION:</b> The Linked List is one of the oldest and most widely used data structures in computer science; plenty of programming languages offer primitives similar to it. The Circularly Linked List is a List that ends up with the beggining of itself, creating a circle. The application of the Circularly Linked List in OWL DL has not been investigated yet. The wikipedia entry offers plenty of information on the subject: http://en.wikipedia.org/wiki/Linked_list.</p><p><b>REFERENCES</b>: <ul><li><a href="http://www.co-ode.org/resources/tutorials/bio/">http://www.co-ode.org/resources/tutorials/bio/</a></li><li>Nick Drummond, Alan Rector, Robert Stevens, Georgina Moulton, Matthew Horridge, Hai H. Wang, Julian Seidenberg. Putting OWL in Order: Patterns for Sequences in OWL. OWLed 2006.</li><li>Robert Stevens, Mikel Egana Aranguren, Katy Wolnstencroft, Ulrike Sattler, Nick Drummond and Mathew Horridge. Using OWL to Model Biological Knowledge. International Journal of Human Computer Studies 2006, 65:7, 583-594.</li></ul></p><p><b>URL</b>: <a href="http://odps.sourceforge.net/odp/owl/Domain_Modelling_ODP/List.owl">http://odps.sourceforge.net/odp/owl/Domain_Modelling_ODP/List.owl</a></p><hr><b>Extension ODPs (by-pass the limitations of OWL):</b> <a href="Nary_DataType_Relationship.html">Nary_DataType_Relationship</a>, <a href="Exception.html">Exception</a>, <a href="Nary_Relationship.html">Nary_Relationship</a>.<br/><b>Good Practice ODPs (obtain a more robust, cleaner and easier to maintain ontology):</b> <a href="Entity_Feature_Value.html">Entity_Feature_Value</a>, <a href="Selector.html">Selector</a>, <a href="Normalisation.html">Normalisation</a>, <a href="Upper_Level_Ontology.html">Upper_Level_Ontology</a>, <a href="Closure.html">Closure</a>, <a href="Entity_Quality.html">Entity_Quality</a>, <a href="Value_Partition.html">Value_Partition</a>, <a href="Entity_Property_Quality.html">Entity_Property_Quality</a>, <a href="DefinedClass_Description.html">DefinedClass_Description</a>.<br/><b>Domain Modelling ODPs (solutions for concrete modelling problems in biology):</b> <a href="Interactor_Role_Interaction.html">Interactor_Role_Interaction</a>, <a href="Sequence.html">Sequence</a>, <a href="CompositePropertyChain.html">CompositePropertyChain</a>, <a href="List.html">List</a>, <a href="Adapted_SEP.html">Adapted_SEP</a>.<hr><a href="index.html">ODPs public catalog</a><hr></body></html>
