<p><b>NAME:</b> Species.</p><p><b>ALSO KNOWN AS:</b> Linnaean taxonomy.</p><p><b>CLASSIFICATION:</b> Domain modelling.</p><p><b>MOTIVATION:</b> Species taxonomy is a very important part of biological knowledge. Yet it is difficult to model directly because, for example, OWL DL does not allow for metaclasses (e.g. Homo as the collection of all Homo organisms and as genera). Another controversial issue is how the bio-entities relate to the actual taxonomy. This ODP attempts to clarify all that by decoupling the organisms hierarchy (the actual individuals) and the taxonomy itself (the taxonomical categories). This is done like that to avoid inconsistent situations like: Mikel is a Homo sapiens, Homo sapiens is a species, so Mikel is a species. We can say, however, that Mikel is an Homo sapiens organism, and that an Homo sapiens organism has the taxonomical level of species..</p><p><b>AIM:</b> To model the traditional Linnaean taxonomy in OWL DL, so biological entities can be linked to it.</p><p><b>STRUCTURE:</b></p><img src="../img/Species_abstract.png"><p><b>SAMPLE:</b></p><img src="../img/Species_instance.png"><p><b>ELEMENTS:</b> The main elements are the Organism hierarchy, with the actual organisms, and the taxonomical levels. These two hierarchies are linked  by the "has_taxonomical_level" property (functional, as a group of organisms can only have one taxonomical level). The organisms are linked by is_sub_organism_of and is_super_organism_of properties, both transitive. This is done like that, instead of using the default is_a relationship, to avoid the inheritance of taxonomical levels (e.g. using the standard is_a HomoSapiensOrganism would end up having the taxonomical level of mammal -class- and homo -genera- as well as species). The properties to link entities to the organisms are two: as_described_in and appears_in. As_described_in is inspired by the "sensu" from the Gene Ontology, meaning an structure as it is described in the given organism. Thus as_described_in should not propagate with is_sub_organism_of or is_super_organism_of. Appears_in means that an entity is actually known to belong to those organisms. As_described_in is a superproperty of appears_in, as anything that is described in an organism also appears in that organism (but not neccesarily in the suborganisms).</p><p><b>IMPLEMENTATION:</b> For each taxonomical level a class is added under TaxonomicalLevel, in a flat list. For each group of organisms a class is added under Organism: each Organism is linked to the corresponding taxonomical level via the property has_taxonomical_level, using universal restrictions. The organisms are linked using existential restrictions in the properties is_sub_organism_of and is_super_organism_of, using a union if a given organism has more than one subgroup. Entities are added to the corresponding organism via restrictions in appears_in and as_described_in properties, using transitive propagation in the case of appears_in, if pertinent (i.e. appears_in some (MammalOrganism or is_sub_organism_of MammalOrganism)).</p><p><b>RESULT:</b> The organism hierarchy and the actual taxonomical levels become decoupled.</p><p><b>RELATED ODPS:</b> AdaptedSEP, Sequence.</p><p><b>ADDITIONAL INFORMATION:</b> The paper by Schulz etal. provides an excellent overview of attempts at defining species and analyses different ODPs for this problem. Moreover, they propose a sound solution (http://www.ifomis.org/biotop) that will be captured soon in this catalogue.</p><p><b>REFERENCES</b>: <ul><li><a href="http://www.odps.sourceforge.net">http://www.odps.sourceforge.net</a></li><li>Stefan Schulz, Holger Stenzhorn, and Martin Boeker. The ontology of biological taxa. Bioinformatics 2008 24: i313-i321; doi:10.1093/bioinformatics/btn158.</li><li>Evaluating ontological decisions with OntoClean. Nicola Guarino, Christopher A. Welty. Communications of the ACM (2002), 45 (2), 61-65.</li></ul></p><p><b>URL</b>: <a href="http://odps.sourceforge.net/odp/owl/Domain_Modelling_ODP/Species.owl">http://odps.sourceforge.net/odp/owl/Domain_Modelling_ODP/Species.owl</a></p><hr><b>Extension ODPs:</b> <a href="Exception.html">Exception</a>, <a href="Nary_DataType_Relationship.html">Nary_DataType_Relationship</a>, <a href="Nary_Relationship.html">Nary_Relationship</a>.<br/><b>Good Practice ODPs:</b> <a href="Entity_Quality.html">Entity_Quality</a>, <a href="Entity_Feature_Value.html">Entity_Feature_Value</a>, <a href="Value_Partition.html">Value_Partition</a>, <a href="DefinedClass_Description.html">DefinedClass_Description</a>, <a href="Normalisation.html">Normalisation</a>, <a href="Upper_Level_Ontology.html">Upper_Level_Ontology</a>, <a href="Closure.html">Closure</a>, <a href="Entity_Property_Quality.html">Entity_Property_Quality</a>.<br/><b>Domain Modelling ODPs:</b> <a href="CompositePropertyChain.html">CompositePropertyChain</a>, <a href="List.html">List</a>, <a href="Adapted_SEP.html">Adapted_SEP</a>, <a href="Interactor_Role_Interaction.html">Interactor_Role_Interaction</a>, <a href="Sequence.html">Sequence</a>, <a href="Species.html">Species</a>.<hr><a href="index.html">ODPs public catalog</a><hr></body></html>